### DISCUSSION TITLE
Copy Playlist as a Collection

### DISCUSSION THREAD

**Discussion by Wetbikeboy2500** [ppy/osu#19051] [Ideas]
Issue: You find a good playlist to play, but there is no quick way to save it locally and long-term.

Solution: Add a copy option to the playlist listings to have them become a local beatmap collection.

I know this is possible as I have created a feature branch [here](https://github.com/Wetbikeboy2500/osu/tree/copy-playlist-collection) implementing this feature. I want to make sure the current flow matches the project direction.

Flow:
Right click a playlist
![image](https://user-images.githubusercontent.com/19276868/178081849-aec123f1-be40-41fc-bc77-21789019a30d.png)
Select copy as collection & a notification is made the the collection was created
![image](https://user-images.githubusercontent.com/19276868/178081912-8a0a3b9a-ff93-4060-86a3-3acf7af6b7a6.png)

It takes all the beatmap hashes the playlist has and adds them to the new collection, so there isn't any mass downloading happening.

One annoyance is how intrusive the notification is which [toasts](https://github.com/ppy/osu/issues/18885) should help.

The other behavior is that a playlist can be copied an infinite number of times with the collections sharing the same name.

**Comment by frenzibyte** [comment]
You mention no mass-downloading, but simply adding the beatmap hashes isn't enough for the collection to be playable locally, and song select offers no UI to show that a beatmap in a collection is missing. Generally resulting in bad UX as the user would be unable to know which beatmaps are actually in the collection.

Would mention https://github.com/ppy/osu/discussions/19019 as an alternative direction.

**Reply by Wetbikeboy2500** [reply]
I do agree with you. The alternative does work, but the user would face a considerable amount of friction trying to save an entire playlist.

It would then be up to communicating better what copying a collection does.

This could be broken into two options:

1 Make an exact copy

All hashes are used and somehow notifying the user that only downloaded maps will appear. This could even be a first time prompt where the user would need to indicate "don't show again" behavior 

2 Make a copy of only downloaded maps

Only allow hashes that are resolvable.


I would prefer having a power-option since I would find it useful to first copy a collection and then start going through it and downloading what I like to play. It leaves no chance of forgetting or having to check if a map is already added to a collection.
